costmap_2d/plugins/static_layer_with_inflation.cpp

```diff
@@ -344,6 +344,47 @@ void StaticLayerWithInflation::updateBounds(double robot_x, double robot_y, doub
   has_updated_data_ = false;
 }
 
+double StaticLayerWithInflation::getDistanceFromStaticMap(double px, double py)
+{
+  if (layered_costmap_->isRolling())
+  {
+    // If it is rolling, convert the points.
+    // If rolling window, the master_grid is unlikely to have same coordinates as this layer
+    unsigned int mx, my;
+    double wx, wy;
+    // Might even be in a different frame
+    tf::StampedTransform transform;
+    try
+    {
+      tf_->lookupTransform(map_frame_, global_frame_, ros::Time(0), transform);
```

Comment:
Double check the api but I believe this may need a waitForTransform?  We had a few issues that if the TF is not available then this call can crash.